src/Button.tsx

```diff
@@ -0,0 +1,291 @@
+
+  /**
+   * Background color to show when button is being hovered over and is not disabled
+   */
+  hoverBackgroundColor?: CSS.BackgroundColorProperty;
+
+  /**
+   * The content to show inside the button
+   *
+   * When using an icon, use the `icon` prop instead.
+   */
+  children?: React.ReactNode;
+
+  /**
+   * Class name to passthrough to the top level component
+   */
+  className?: string;
+
+  /**
+   * Disable the button
+   *
+   * This will prevent all clicks and will show the button in a different color
+   */
+  disabled?: boolean;
+
+  /**
+   * Background color to display when the button is disabled
+   */
+  disabledBackgroundColor?: CSS.BackgroundColorProperty;
+
+  /**
+   * Text color to display when the button is disabled
+   */
+  disabledTextColor?: CSS.BackgroundColorProperty;
+
+  /**
+   * Treat this button as a floating action button
+   *
+   * You must include an `icon` prop and you must _not_ include a `children`
+   * prop for a floating action button.
+   */
+  fab?: boolean;
+
+  /**
+   * Force the button to be in the active state.
+   *
+   * Useful for Storybook
+   *
+   * @ignore
+   */
+  forceActiveState?: boolean;
+
+  /**
+   * Force the button to be in the focus state.
+   *
+   * Useful for Storybook
+   *
+   * @ignore
+   */
+  forceFocusState?: boolean;
+
+  /**
+   * Force the button to be in the hover state.
+   *
+   * Useful for Storybook
+   *
+   * @ignore
+   */
+  forceHoverState?: boolean;
+
+  /**
+   * A hidden button will have no background
+   *
+   * You should configure the `hiddenActiveBackgroundColor` prop if you are
+   * using any custom colors.
+   */
+  hidden?: boolean;
+
+  /**
+   * Color to make the background for a hidden button that is active.
+   *
+   * This color should be darker than the `hoverBackgroundColor`.
+   *
+   * This should not be used for a non-hidden button.
+   */
+  hiddenActiveBackgroundColor?: CSS.BackgroundColorProperty;
+
+  /**
+   * Color to make the text for a focused hidden button.
+   *
+   * This should not be used for a non-hidden button.
+   */
+  hiddenFocusedColor?: CSS.ColorProperty;
+
+  /**
+   * Either an icon to show to the left of the button text, or on it's own
+   */
+  icon?: React.ReactElement;
+
+  /**
+   * Size of the button
+   *
+   * Defaults to "default"
+   */
+  size?: "default" | "small" | "large";
+
+  /**
+   * CSS styles to apply
+   *
+   * @ignore
+   */
+  style?: React.CSSProperties;
+}
+
+/**
+ * Style system for Space Kit buttons
+ *
+ * This is intended to be used as an abstraction for your project's style guide.
+ * There are way too many configuration options to use this component directly.
+ *
+ * @see https://zpl.io/amdN6Pr
+ */
+export const Button: React.FC<Props> = ({
+  backgroundColor = colors.silver.light,
+  children,
+  disabled = false,
+  disabledBackgroundColor = colors.silver.dark,
+  disabledTextColor = colors.grey.light,
+  fab = false,
+  forceActiveState = false,
```

Comment:
We should consider if we want to be exposing the different button states in Storybook, or if displaying a single interactive button for each variant and size is enough. It might be nice to not need to support these `force*` props

Reply:
A bit surprised to see these. I recall when we discussed this over the kitchen table, you were a bit perplexed about why I was considering setting active state through props as part of the button's API, and wanted to forego that until we needed this in any project

I'm guessing then that these are solely for the purpose of being able to show them in storybook. I agree that there is value in having that to allow  visual regression tests. I don't think adding public props solely for storybook-ability is a good idea. E.g. when I'm using vscode's code completion to see what a components props are, I don't want to see props that I'm not supposed to use

Let's use class names here instead. We should be able to achieve the same result without exposing test-apis as props